src/utils/HealthLogger.js

```diff
@@ -0,0 +1,169 @@
+    }
+
+    /**
+     * Return the Performance related data
+     * @returns {Object} Performance Data aggregated till now
+     */
+    function getPerformanceData() {
+        return PerfUtils.getHealthReport();
+    }
+
+    /**
+     * Return the aggregate of all health data logged till now from all sources
+     * @returns {Object} Health Data aggregated till now
+     */
+    function getAggregatedHealthData() {
+        var healthData = PreferencesManager.getViewState(HEALTH_DATA_STATE_KEY);
+        jQuery.extend(healthData, getPerformanceData());
+        return healthData;
+    }
+
+    /**
+     * Return all health data logged till now stored in the state prefs
+     * @returns {Object} Health Data aggregated till now
+     */
+    function getHealthData() {
+        return PreferencesManager.getViewState(HEALTH_DATA_STATE_KEY);
+    }
+
+    /**
+     * Sets the health data
+     * @param {Object} dataObject The object to be stored as health data
+     */
+    function setHealthData(dataObject) {
+        if (!shouldLogHealthData()) {
+            return;
+        }
+        PreferencesManager.setViewState(HEALTH_DATA_STATE_KEY, dataObject);
+    }
+
+    /**
+     * Returns health data logged for the given key
+     * @returns {Object} Health Data object for the key or undefined if no health data stored
+     */
+    function getHealthDataLog(key) {
+        var healthData = getHealthData();
+        return healthData[key];
+    }
+
+    /**
+     * Sets the health data for the given key
+     * @param {Object} dataObject The object to be stored as health data for the key
+     */
+    function setHealthDataLog(key, dataObject) {
+        var healthData = getHealthData();
+        healthData[key] = dataObject;
+        setHealthData(healthData);
+    }
+
+    /**
+     * Clears all the health data recorded till now
+     */
+    function clearHealthData() {
+        PreferencesManager.setViewState(HEALTH_DATA_STATE_KEY, {});
+        //clear the performance related health data also
+        PerfUtils.clear();
+    }
+
+    /**
+     * Enable or disable health data logs
+     * @param {Boolean} enabled true to enable health logs
+     */
+    function setHealthLogsEnabled(enabled) {
+        logHealthData = enabled;
+        if (!enabled) {
+            clearHealthData();
+        }
+    }
+
+    /**
+     * When ever a file is opened call this function. The function will record the number of times
+     * the standard file types have been opened. We only log the standard filetypes
+     * @param {String} filePath          The path of the file to be registered
+     * @param {Boolean} addedToWorkingSet set to true if extensions of files added to the
+     *                                    working set needs to be logged
+     */
+    function fileOpened(filePath, addedToWorkingSet) {
+        if (!shouldLogHealthData()) {
+            return;
+        }
+        var fileExtension = FileUtils.getFileExtension(filePath),
```

Comment:
I wonder if it makes sense to also log multiple file extensions, like in `.min.js` and `.min.css` (which are definitely the most common).
It'd be an interesting metric to see how many of the opened files are _minified_.